crates/miden-objects/src/batch/proposed_batch.rs

```diff
@@ -0,0 +1,42 @@
+use alloc::{sync::Arc, vec::Vec};
+
+use crate::{note::NoteInclusionProofs, transaction::ProvenTransaction};
+
+/// A proposed batch of transactions with all necessary data to validate it.
+#[derive(Debug, Clone)]
+pub struct ProposedBatch {
```

Comment:
`ProposedBatch` and `ProvenBatch` sound nice together, but I wonder, for consistency with other input types like `TransactionInputs` or `BlockInputs`, whether this should actually be called `BatchInputs`?

Reply:
Yeah, not sure about renaming `TransactionInputs` either. That seems fine.
`ProposedBlock` sounds good as well, but I guess we can see what to do with it when working on the block kernel. For now, I'll just leave `ProposedBatch` as-is.